sql/sql_insert.cc

```diff
@@ -1948,6 +1948,8 @@ int write_record(THD *thd, TABLE *table,COPY_INFO *info)
           if (likely(!error))
           {
             info->deleted++;
+            if (!table->file->has_transactions())
```

Comment:
No `versioned()` check. Did you run full test suite?

Reply:
It can be without system versioning, but it's extremely hard to catch.
One possible scenario is:
1. REPLACE two rows in one command
2. wait until first row replace is succeeded (it should go through `ha_update_row` branch, e.g. the just fixed one)
3. wait until `ha_write_row` is failed in while() condition
4. physically unplug the data storage (table files can be locked on fs level, I suppose)

So the thing is to make second `ha_update_row` fail. I did not succeed in it, so here's the versioning test case.

Reply:
I have to run it, if you modify non-versioning code.

Reply:
s/I/You/

Reply:
http://buildbot.askmonty.org/buildbot/grid?category=main&branch=tt-10.3-MDEV-17891

Reply:
What is the status of this task? If tests have passed, please merge and close.